### PR TITLE
Update to Swift 4.1

### PR DESCRIPTION
**Pull request by ShihabMehboob**
Update to swift 4.1 in reference to this [PR](https://github.ibm.com/arf/planning-swift-solutions/issues/571)

### CODE REVIEW

**Comment by billabt** [comment]
Can't merge this until CI is passing...  Thanks.

File: .swift-version [hunk 1]
```diff
@@ -1 +1 @@
-3.1.1
+4.1
```
Comment (djones6) [inline .swift-version]: We can't set this until 4.1 is released (the build doesn't exist).
If we want to test with 4.1 before release (a *good idea*), then we should add another entry to Travis to test with the latest 4.1 development snapshot (for both Linux and Mac).

File: .travis.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ matrix:
     - os: osx
       osx_image: xcode9.2
       sudo: required
-      env: SWIFT_SNAPSHOT=4.0      
+      env: SWIFT_SNAPSHOT=4.2
```
Comment (djones6) [inline .travis.yml]: See my comment above (and I'm assuming '4.2' was a typo?).  This should stay at 4.0 for now, but do add another OSX entry for the latest 4.1 snapshot.

File: Package.resolved [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        "state": {
+          "branch": null,
+          "revision": "0d2663bf81ae2588b61b358fe4a25d6f0dc4d8af",
+          "version": "0.12.87"
+        }
+      }
+    ]
+  },
+  "version": 1
+}
```
Comment (djones6) [inline Package.resolved]: I don't think you meant to check this file in?

File: Package.swift [hunk 4]
```diff
@@ -4,40 +4,39 @@
...
+#endif
+
 #else
-	
-	fatalError("Unsupported OS")
-	
+
+fatalError("Unsupported OS")
+
 #endif
```
Comment (djones6) [inline Package.swift]: I'd avoid checking in whitespace-only changes (unless there's a *real* change in here that I've not spotted?).  It makes *real* changes hard to spot :)

File: Package@swift-4.swift [hunk 5]
```diff
@@ -62,4 +60,3 @@ let package = Package(
 fatalError("Unsupported OS")
 
 #endif
-
```
Comment (djones6) [inline Package@swift-4.swift]: Same comment as above

File: .swift-version (hunk 1 above)
Comment (billabt) [reply .swift-version]: I agree.

File: .travis.yml (hunk 2 above)
Comment (billabt) [reply .travis.yml]: Yup, totally agree.

File: Package.resolved (hunk 3 above)
Comment (billabt) [reply Package.resolved]: Agreed, this file never should've been checked in.

**Comment by billabt** [comment]
I'm going to _reject_ this PR.  It should be redone properly.

**Comment by billabt** [comment]
I'll be rejecting a similar PR for [BlueSocket](https://github.com/IBM-Swift/BlueSocket/pull/116) for the same reasons.

**Comment by billabt** [comment]
In this PR and the one against _BlueSocket_ there were many totally irrelevant and unnecessary changes made.  Please keep the changes restricted to the _minimum_ required to accomplish what you're trying to do.

**Pull request closed without merging**